test/integration/enterprise_test.go

```diff
@@ -67,77 +73,127 @@ func TestKongEnterprisePostgres(t *testing.T) {
+	httpBinAddon := httpbin.New()
+	require.NoError(t, env.Cluster().DeployAddon(ctx, httpBinAddon))
 	require.NoError(t, <-env.WaitForReady(ctx))
 
 	t.Log("accessing httpbin via ingress to validate that the kong proxy is functioning")
-	httpc := http.Client{Timeout: time.Second * 10}
-	require.Eventually(t, func() bool {
-		resp, err := httpc.Get(fmt.Sprintf("%s/%s", proxyURL, httpbinAddon.Path()))
-		if err != nil {
-			t.Logf("WARNING: error while waiting for %s: %v", proxyURL, err)
-			return false
-		}
-		defer resp.Body.Close()
-		if resp.StatusCode == http.StatusOK {
-			b := new(bytes.Buffer)
-			n, err := b.ReadFrom(resp.Body)
-			require.NoError(t, err)
-			require.True(t, n > 0)
-			return strings.Contains(b.String(), "<title>httpbin.org</title>")
-		}
-		return false
-	}, time.Minute*3, time.Second)
+	req, err = http.NewRequestWithContext(
+		ctx, http.MethodGet,
+		proxyURL.JoinPath(httpBinAddon.Path()).String(),
+		nil,
+	)
+	require.NoError(t, err)
+	eventuallyExpectedStatusCodeAndBody(t, httpClient, req, http.StatusOK, func(t *testing.T, body []byte) bool {
+		const expectedBody = "<title>httpbin.org</title>"
+		t.Logf("check expected content %s of the response body", expectedBody)
+		return bytes.Contains(body, []byte(expectedBody))
+	})
+
+	const workspaceToCreate = "test-workspace"
+	if adminPassword != "" {
+		t.Log("verifying enterprise workspace API functionality using /workspaces (works only for dbmode)")
+		req, err = http.NewRequestWithContext(
+			ctx, http.MethodPost, adminURL.JoinPath("/workspaces").String(),
+			strings.NewReader(fmt.Sprintf(`{"name": "%s"}`, workspaceToCreate)),
+		)
+		require.NoError(t, err)
+		decorateRequestWithAdminPassword(t, req, adminPassword)
+	} else {
+		t.Log("verifying enterprise workspace API functionality using /config (works only for dblessmode)")
+		t.Fatal("not implemented yet")
+	}
+	req.Header.Set("Content-Type", "application/json")
+	eventuallyExpectedStatusCodeAndBody(t, httpClient, req, http.StatusCreated, nil)
+
+	t.Log("verifying that the workspace was indeed created")
+	req, err = http.NewRequestWithContext(
+		ctx, http.MethodGet,
+		adminURL.JoinPath("/workspaces/").JoinPath(workspaceToCreate).String(),
+		nil,
+	)
+	require.NoError(t, err)
+	if adminPassword != "" {
+		decorateRequestWithAdminPassword(t, req, adminPassword)
+	}
+	eventuallyExpectedStatusCodeAndBody(t, httpClient, req, http.StatusOK, nil)
+
+}
+
+func decorateRequestWithAdminPassword(t *testing.T, req *http.Request, adminPassword string) {
+	t.Helper()
+	const passwdHeader = "Kong-Admin-Token"
+	t.Logf("setting the admin api password (Kong-Admin-Token header %s)", passwdHeader)
+	req.Header.Set(passwdHeader, adminPassword)
+}
+
+func prepareKongEnterpriseLicense(t *testing.T) string {
+	t.Helper()
+	t.Log("preparing kong enterprise license")
+	licenseJSON, err := kong.GetLicenseJSONFromEnv()
+	require.NoError(t, err)
+	return licenseJSON
+}
+
+func eventuallyExpectedStatusCodeAndBody(
```

Comment:
Can you consider putting this in a place where it can be reused (not only by integration tests) e.g. in https://github.com/Kong/kubernetes-ingress-controller/blob/dcd58c38c4b53689a863e6a0e67d4684ab37df13/test/internal/helpers/http.go and exporting it?

Reply:
Good idea, I'm doing it

Reply:
But, you're referencing the KIC repo, and here is KTF. The place where can I put it (and it can be reused by everything that imports KTF) is [pkg/utils](https://github.com/Kong/kubernetes-testing-framework/tree/c043d71850708036bf205f0d0553b5b5661bbb32/pkg/utils). WDYT?

Reply:
No `utils` please 🙃 We can introduce a new package which will aggregate testing types and functions. Could be placed under `pkg/test/` (this project is named `*-testing-framework` after all).

Ideally we'd think twice about the API so that it could be reused and not be too specific. E.g. introduce option parameters that could be applied on the HTTP response etc.

Reply:
Yeah, but now I'm wondering if helpers tight to the specific testing library should be part of the public API of the package...

Reply:
Do you have any other suggestions? This can be reused in many places e.g. KGO, KIC etc. We could theoretically use a separate repo for those but I don't see a reason at this stage.

Perhaps we would go this route if several of those "test helpers" were moved (from e.g. KIC) to said repo and imported in KIC and other projects.

Reply:
Nothing concrete. I have the same feeling as you and see the benefit. I think we should discuss it with the team and go through all repos and check what we have already and where/how we want to put it. And for this PR, leaving it as is for now

Reply:
Ok agreed. Let's create an issue: can be in e.g. team-k8s repo and discuss this on next team sync.

One point that I'd prefer to resolve here still is the API of this: the interface requires the user to 
- provide a `nil` argument when body checker param is unused. This I believe could be improved by aforementioned options pattern.
- keep rewriting the checks that most likely will be reused. We could also provide the option to e.g. check if the version is enterprise (as used above) and reuse that, making the call site that much cleaner.

Last but not least: I'd suggest to put this in a separate package but given the unresolved debate above we can make this an `internal` package for now, so that we can reuse it in KTF but we don't have to worry about breaking users until we expose it (by placing it in a non `internal` package).

WDYT?

Reply:
I provided opts pattern and introduced an internal package. For checking if it's an enterprise after a couple of tries I couldn't come up with a clean API, so I left it as it is. The issue has been created https://github.com/Kong/team-k8s/issues/292